Transactions

tsohleDev opened the pull request:
**Query with transactions**

- Add a column to the table
- Add more data
- Make queries in form of transactions

tsohleDev committed:
create table

tsohleDev committed:
Add the data

tsohleDev committed:
Add queries,screenshot and README file

tsohleDev committed:
update table, insert a column

tsohleDev committed:
Add data

tsohleDev committed:
Add queries with their resullt screenshots

danmainah commented on README.md:
```diff
@@ -0,0 +1,198 @@
+
+# 📖 [Vet Clinic] <a name="about-project"></a>
+
+> Vet clinic is a tutorial on how to create a relational database in SQL, where we create a vet clinic's database
```
- [x] @tsohleDev kindly remove the guideline statements, they are supposed to only guide you as you create the readme file

tsohleDev replied:
I see, thank you for the info

danmainah commented on data.sql:
```diff
@@ -0,0 +1,22 @@
+/*Part Two*/
+INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_kg, species)
+VALUES 
+  (4,'Charmander', '2020-02-08', 0, false, 11, 'lizard'),
+  (6,'Plantmon', '2021-11-15', 2, true, 5.7, 'plant'),
+  (7,'Squirtle', '1993-04-02', 3, false, 12.13, 'turtle'),
+  (8,'Angemon', '2005-06-12', 1, true, 45, 'angel'),
+  (9,'Boarmon', '2005-06-07', 7, true, 20.4, 'boar'),
+  (10,'Blossom', '1998-10-13', 3, true, 17, 'flower'),
+  (11,'Ditto', '2022-05-14', 4, true, 22, 'transformer');
+
```
- [x] @tsohleDev kindly ensure you have only data provided in the project, remove the species data, and also the weights are correct, some are supposed to be negative.

danmainah commented on queries.sql:
```diff
@@ -0,0 +1,104 @@
+/*Part Two*/
+-- UPDATE SPECIES WITH ROLLBACK
+BEGIN;
+
+UPDATE animals
+SET species = 'unspecified';
+
+ROLLBACK;
+
```
- [x] @tsohleDev kindly add code to  verify that change was made before and after rollback eg ``SELECT species from animals;``

tsohleDev replied:
yes indeed

afanimartin replied:
- Please make the requested change. Run a SELECT query after the rollback

danmainah commented on queries.sql:
```diff
@@ -0,0 +1,104 @@
+ROLLBACK;
+
+-- UPDATE SPECIES WITHOUT ROLLBACK 
+BEGIN TRANSACTION;
+
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+
+COMMIT;
+
```
- [x]  @tsohleDev kindly add code to verify that the change was made before and after the commit.

danmainah commented on queries.sql:
```diff
@@ -0,0 +1,104 @@
+
+-- DELETE ALL RECORDS AND ROLLBACK
+BEGIN;
+
+DELETE FROM animals;
+
+ROLLBACK;
+
+-- UPDATE ANIMALS
```
- [x]  @tsohleDev kindly add code to verify that the change was made before and after the rollback

afanimartin replied:
- Please make the requested change. Run a SELECT query after the rollback

tsohleDev replied:
I actually did, you just looking at the outdated part

danmainah reviewed (changes requested):
Hi @tsohleDev , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Gitflow was correctly used
✔️  Good design


# Required Changes ♻️

- [ ] Kindly add screenshots of the queries results to the pull request description, and kindly ensure the result are correct

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

tsohleDev committed:
Amend README and queries

remove a project description
Add queries to show the changes that are made by the transections

afanimartin commented on README.md:
```diff
@@ -0,0 +1,196 @@
+# curriculum-databases-projects-template
+
+> This template should be used for database related projects at Microverse.
+> Generate your own repository, update this README and edit all files content while working on projects. You should not be adding any new files unless asked otherwise.
```
- Remove this guidelines as suggested. Add content relevant to the section instead. Do the same to all sections.

afanimartin reviewed (changes requested):
Hi @tsohleDev, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## Highlights 👍🏿 👍🏿 👍🏿 
- Prompt communication with reviewers

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

### Martin Afani
#### Slack: whiz25

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

tsohleDev committed:
Update README.md

tsohleDev committed:
Add quiries

tsohleDev committed:
Merge branch 'transactions' of https://github.com/tsohleDev/vet-clinic into transactions

Kossi-stack reviewed (approved):
## Status: Approved ✔️

**Hi @tsohleDev,**

You did a really great job! 💯 💯💯💯
Congratulations! 🎉 🎉 🎉 🎉 

![128382030-38e7de91-f6f2-45cf-b54e-8068e0446e4c](https://user-images.githubusercontent.com/81444338/150871274-7d1b779c-88e6-420b-a530-b7b905eaa3ac.gif)

You may close this Pull Request and merge the changes
Keep being awesome dev's and good luck with your next milestone!

## To Highlight

✅ Descriptive pull request message
✅ Documented your work professionally
✅ README file is descriptive.
✅ Commit messages are meaningful.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding! 👏 👏 👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
